mutation_partition.cc

```diff
@@ -2074,22 +2074,26 @@ void query_result_builder::consume_new_partition(const dht::decorated_key& dk) {
 
 void query_result_builder::consume(tombstone t) {
     _mutation_consumer->consume(t);
+    _stop = _rb.bump_and_check_tombstone_limit();
 }
 stop_iteration query_result_builder::consume(static_row&& sr, tombstone t, bool is_live) {
     if (!is_live) {
+        _stop = _rb.bump_and_check_tombstone_limit();
         return _stop;
     }
     _stop = _mutation_consumer->consume(std::move(sr), t);
     return _stop;
 }
 stop_iteration query_result_builder::consume(clustering_row&& cr, row_tombstone t,  bool is_live) {
     if (!is_live) {
+        _stop = _rb.bump_and_check_tombstone_limit();
```

Comment:
Suppose bump_and_check returns no. Won't it override a yes in _stop?

Reply:
`_stop` is always `::no` here. After one of the `consume()` methods setting it to `::yes`, the consume loop is immediately broken and `consume_end_of_partition()` is called next.

Reply:
Maybe we shouldn't have a variable _stop then.

Reply:
(e.g. it shoudn't be a member, just a local)

Reply:
We need it as a member, because we need to remember whether we stopped do to us (us=`query_result_builder`) requesting the stop, or it was the layer above us. In the former case the read is declared short.

Reply:
I see. Maybe a comment.

Reply:
added comment